### PR TITLE
Add support to create Clusters with ADLS as default Storage

### PR DESCRIPTION
**Pull request by vahemesw**
- Add support to create Clusters with ADLS as default Storage

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vahemesw__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (vahemesw). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/hdinsight/hdiCustomization._js [hunk 1]
```diff
@@ -190,6 +192,80 @@ __.extend(HdiCustomization.prototype, {
...
+        storageAccountNameKey = 'fs.default.name';
+      }    
+      var container = utils.stringIsNullOrEmpty(storageAccountDetails.defaultStorageContainer) ?
+        clusterName : storageAccountDetails.defaultStorageContainer;
+      coreConfig[storageAccountNameKey] = util.format($('wasb://%s@%s'), container, storageAccountDetails.defaultStorageAccountName);
+    }
+
+    var defaultStorageConfigKey = util.format($('fs.azure.account.key.%s'), storageAccountDetails.defaultStorageAccountName);
+    var defaultStorageAccount = coreConfig[defaultStorageConfigKey];
+    if (defaultStorageAccount === null || defaultStorageAccount === undefined) {
```
Comment (balajikris) [inline lib/commands/arm/hdinsight/hdiCustomization._js]: `if(!defaultStorageAccount)` , you get the idea. please update other locations after this as well.

File: lib/commands/arm/hdinsight/hdiCustomization._js [hunk 2]
```diff
@@ -190,6 +192,80 @@ __.extend(HdiCustomization.prototype, {
...
 
+  getCoreSiteConfigurationsForWASB : function(configurations, clusterName, hdiVersion, storageAccountDetails) {
+    var coreConfig = configurations[HDIConstants.ConfigurationKey.CoreSite];
+    if (coreConfig === null || coreConfig === undefined) {
+      coreConfig = {};
+    }
+    var defaultFS = coreConfig['fs.defaultFS'];
+    if (defaultFS === null || defaultFS === undefined) {
+      var storageAccountNameKey = 'fs.defaultFS';
+      if (hdiVersion !== null && hdiVersion === '2.1') {
```
Comment (vahemesw) [inline lib/commands/arm/hdinsight/hdiCustomization._js]: hdiVersion cannot be undefined. Default value is set on this parameter before the code flow reaches this particular stage.

**Comment by balajikris** [comment]
Looks good to me. @giyer - would you like to take a look as well or shall I go ahead and merge? 
/cc: @vahemesw

**Comment by vahemesw** [comment]
giyer approved these changes here - http://codeflow/Dashboard/review/display/vahemesw-e009af64403b4f2892febdd83b04aa50. Can you please go ahead and merge the changes?

**Pull request merged**